docs/amber_script.md

```diff
@@ -405,6 +403,6 @@ CLEAR_COLOR kGraphicsPipeline 255 0 0 255
 CLEAR kGraphicsPipeline
 
 RUN kGraphicsPipeline \
-  DRAW_ARRAY kIndices IN kBuffer AS triangle_list \
+  DRAW_ARRAY kIndices IN kData AS triangle_list \
```

Comment:
Currently RUN is where we use vertex and indices buffers, do they need to be attached into a PIPELINE with the VERTEX_BUFFER and INDEX_BUFFER calls we mentioned, or is just referencing them here enough?

If they're attached into a PIPELINE, does that mean that `RUN pipeline DRAW_ARRAY` will just use them implicitly? So this would just be:

```
 RUN kGraphicsPipeline DRAW_ARRAY AS triangle_list START_IDX 0 COUNT 24
```

Reply:
In Vulkan both index buffer and vertex buffers are bound in the command buffer before doing the draw call.  So it's ok to do it implicitly here.  The backend will translate this RUN call into the right bind-index-buffer and bind-vertex-buffer commands.    

I believe the vertex and index buffer bindings are totally different numbering spaces than descriptor set/binding pairs.  Beware.(?!)